Restructure Project

saddieeiddas opened the pull request:
I have put together an updated structure for the project.

**NOTE:** This might not be the desired structure, so hopefully this PR will allow for some discussion.

You can see the branch here: https://github.com/saddieeiddas/Camelot-Unchained-Client-Library/tree/dev

##### `src` now looks like this:

![new-src-structure](https://cloud.githubusercontent.com/assets/8982798/11770839/179fea98-a200-11e5-8bc4-f747b0f53a56.png)

##### `lib` now looks like this:

![new-lib-structure](https://cloud.githubusercontent.com/assets/8982798/11770836/115a9f98-a200-11e5-875b-5b04e32f82e4.png)


##### `dist` now looks like this:

![new-dist-structure](https://cloud.githubusercontent.com/assets/8982798/11770832/0bbd2c2c-a200-11e5-9272-d091c69c34d9.png)

---

I would also like to change the javascript exports for the library, so that all core classes are accessed through a `core.*` namespace, instead of being exported by `main.ts` directly.
I have added this, but kept the existing exports too to prevent BC breaks.

---

This needs some discussion to see if this structure will simplify things for us.

It also needs to be tested out on different modules, to ensure things don't break.

review-ninja commented:
<a href="https://app.review.ninja/CUModSquad/Camelot-Unchained-Client-Library/pull/9" target="_blank"><img src="https://app.review.ninja/47362620/pull/9/badge" alt="ReviewNinja"/>

saddieeiddas commented:
This is now updated to fix conflicts.

I have also removed the `core.*` exporting, so all exports should continue to be called as they are currently.

Also fixed some issues with styles.

saddieeiddas commented:
have tested this now:

everything builds successfully.

running in game, it seems as if all existing cu-ui components are working as intended
